pkg/registry/apis/query/client/testdata.go

```diff
@@ -1,59 +1,57 @@
-package runner
+package client
 
 import (
 	"context"
 	"fmt"
+	"net/http"
 	"time"
 
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
-
+	data "github.com/grafana/grafana-plugin-sdk-go/experimental/apis/data/v0alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 
-	"github.com/grafana/grafana/pkg/apis/query/v0alpha1"
+	query "github.com/grafana/grafana/pkg/apis/query/v0alpha1"
 	testdata "github.com/grafana/grafana/pkg/tsdb/grafana-testdata-datasource"
 	"github.com/grafana/grafana/pkg/tsdb/legacydata"
 )
 
 type testdataDummy struct{}
 
-var _ v0alpha1.QueryRunner = (*testdataDummy)(nil)
-var _ v0alpha1.DataSourceApiServerRegistry = (*testdataDummy)(nil)
+var _ data.QueryDataClient = (*testdataDummy)(nil)
+var _ query.DataSourceApiServerRegistry = (*testdataDummy)(nil)
 
-// NewDummyTestRunner creates a runner that only works with testdata
-func NewDummyTestRunner() v0alpha1.QueryRunner {
+// NewTestDataClient creates a runner that only works with testdata
+func NewTestDataClient() data.QueryDataClient {
 	return &testdataDummy{}
 }
 
-func NewDummyRegistry() v0alpha1.DataSourceApiServerRegistry {
+// NewTestDataRegistry returns a registry that only knows about testdata
+func NewTestDataRegistry() query.DataSourceApiServerRegistry {
 	return &testdataDummy{}
 }
 
 // ExecuteQueryData implements QueryHelper.
-func (d *testdataDummy) ExecuteQueryData(ctx context.Context,
-	// The k8s group for the datasource (pluginId)
-	datasource schema.GroupVersion,
-
-	// The datasource name/uid
-	name string,
-
-	// The raw backend query objects
-	query []v0alpha1.GenericDataQuery,
-) (*backend.QueryDataResponse, error) {
-	if datasource.Group != "testdata.datasource.grafana.app" {
-		return nil, fmt.Errorf("expecting testdata requests")
-	}
-
-	queries, _, err := legacydata.ToDataSourceQueries(v0alpha1.GenericQueryRequest{
-		Queries: query,
-	})
+func (d *testdataDummy) QueryData(ctx context.Context, req data.QueryDataRequest) (int, *backend.QueryDataResponse, error) {
+	queries, _, err := legacydata.ToDataSourceQueries(req)
 	if err != nil {
-		return nil, err
+		return http.StatusBadRequest, nil, err
 	}
 
-	return testdata.ProvideService().QueryData(ctx, &backend.QueryDataRequest{
-		Queries: queries,
-	})
+	code := http.StatusOK
+	qdr := &backend.QueryDataRequest{Queries: queries}
+	rsp, err := testdata.ProvideService().QueryData(ctx, qdr)
+	if err == nil {
+		for _, v := range rsp.Responses {
+			if v.Error != nil {
+				code = http.StatusMultiStatus
+				break
+			}
+		}
+	} else {
```

Comment:
this logic is used in a couple places. should it be a function?

Reply:
done